src/Compilers/CSharp/Test/Symbol/Symbols/ModuleInitializersTests.cs

```diff
@@ -139,6 +141,131 @@ namespace System.Runtime.CompilerServices { class ModuleInitializerAttribute : S
 Program.Main");
         }
 
+        [Fact]
+        public void SingleCallIsGeneratedWhenMethodIsMarkedTwice()
+        {
+            string source = @"
+using System;
+using System.Runtime.CompilerServices;
+
+class C
+{
+    [ModuleInitializer, ModuleInitializer]
+    internal static void M() => Console.WriteLine(""C.M"");
+}
+
+class Program 
+{
+    static void Main() { }
+}
+
+namespace System.Runtime.CompilerServices
+{
+    [AttributeUsage(AttributeTargets.Method, AllowMultiple = true)]
+    class ModuleInitializerAttribute : System.Attribute { } 
+}
+";
+            CompileAndVerify(source, parseOptions: s_parseOptions, expectedOutput: "C.M");
+        }
+
+        [Fact]
+        public void AttributeCanBeAppliedWithinItsOwnDefinition()
+        {
+            string source = @"
+using System;
+
+class Program 
+{
+    static void Main() => Console.WriteLine(""Program.Main"");
+}
+
+namespace System.Runtime.CompilerServices
+{
+    class ModuleInitializerAttribute : System.Attribute 
+    { 
+        [ModuleInitializer]
+        internal static void M() => Console.WriteLine(""ModuleInitializerAttribute.M"");
+    } 
+}
+";
+            CompileAndVerify(source, parseOptions: s_parseOptions, expectedOutput: @"
+ModuleInitializerAttribute.M
+Program.Main");
+        }
+
+        [Fact]
+        public void ExternMethodCanBeModuleInitializer()
+        {
+            string source = @"
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+class C
+{
+    [ModuleInitializer, DllImport(""dllName"")]
+    internal static extern void M();
+}
+
+namespace System.Runtime.CompilerServices { class ModuleInitializerAttribute : System.Attribute { } }
+";
+            CompileAndVerify(source, parseOptions: s_parseOptions);
```

Comment:
perhaps we should verify IL in `<Module>.cctor` in this scenario?

Reply:
Method body IL or just the presence of the cctor?

Reply:
Guessing just the presence of the cctor, but I'm happy to do more.